# script to call genotypes in can x brev from read counts
#2019-06-26

#set wd to "ancestry" folder generated by assing.ancestry.loop.sh
setwd("~can.x.brev/ancestry")

#read in all of the ancestry ouput files
file.list.ancestry = list.files(pattern="output")
ind.list = gsub("output.", "", file.list.ancestry)
ind.list = gsub(".txt", "", ind.list)

data.list = lapply(file.list.ancestry, read.table)


# read in marker data, e.g., bin names, chromosome, and bin edge positions from marker.conversion.txt
mark.con=read.table("~can.x.brev/marker.conversion.txt", head=T, sep="\t")
mark.con.s=split(mark.con, mark.con$chrom)

# change working dir to print tables in new place
setwd("~can.x.brev/allele.freq.tables")

# make a loop to cycle through individuals
for (ind in 1:length(data.list)){
#for (ind in 2:10){ #i uncomment this and comment above line to test that the script is working

	ind.data = data.list[[ind]]
	ind.data.s=split(ind.data, ind.data$V1)
	table.out <- data.frame()

	# loop for each chromosome

	for (i in 1:7){

	bin.mk <- mark.con.s[[i]][,2]
	bin.min <- mark.con.s[[i]][,3]
	bin.max <- mark.con.s[[i]][,4]
	chrom <- rep(i, length(bin.mk))
	allele.freqs <- rep(NA, length(bin.mk))
	chrom.data <- ind.data.s[[i]]


	# loop for each position on the chromosome
		
		for (cur.bin in 1:length(bin.mk)) { 
	
		bin.min.x <- bin.min[cur.bin] #Figure out what the x-coordinate of the left end of the current bin is
		bin.max.x <- bin.max[cur.bin]
		
		#Extract a data frame that has only the SNPs within our bin
		cur.data <- chrom.data[(chrom.data$V2 >= bin.min.x) & (chrom.data$V2 < bin.max.x),]
		
		#Count how many total 'can' alleles we found in this region
		can.count <- sum(cur.data$V3)
		
		#Count how many total 'brev' alleles we found in this region
		brev.count <- sum(cur.data$V4)
		
		#Figure out the frequency of the 'can' allele
		can.freq <- can.count / (can.count + brev.count)
		
		#Store it in the appropriate slot of allele.freqs
		allele.freqs[cur.bin] <- can.freq 
		
		}
		
		table.out <- rbind(table.out, data.frame(chrom, bin.mk, allele.freqs))
	
	}

write.table(table.out, file=paste(ind.list[[ind]], ".can.freq.txt", sep=""), quote=F, row.names=F, col.names=F)

}